src/wazuh_modules/agent_upgrade/manager/wm_agent_upgrade_tasks_callbacks.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Wazuh Module for Agent Upgrading
+ * Copyright (C) 2015-2020, Wazuh Inc.
+ * July 3, 2020.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+#include "wazuh_modules/wmodules.h"
+#include "wm_agent_upgrade_tasks.h"
+#include "wm_agent_upgrade_parsing.h"
+#include "wm_agent_upgrade_validate.h"
+#include "os_net/os_net.h"
+#include "shared.h"
+
+int wm_agent_upgrade_task_module_callback(cJSON *json_response, const cJSON* task_module_request, cJSON* (*success_callback)(int *error, cJSON* input_json), void (*error_callback)(int agent_id)) {
+    int agents = 0;
+    int error = OS_SUCCESS;
+    cJSON *task_module_response = NULL;
+    cJSON *error_json = NULL;
+
+    if (agents = cJSON_GetArraySize(task_module_request), !agents) {
+        return OS_INVALID;
+    }
+
+    // Send request to task module
+    task_module_response = wm_agent_upgrade_send_tasks_information(task_module_request);
+
+    if (task_module_response && (task_module_response->type == cJSON_Array) && (agents == cJSON_GetArraySize(task_module_response))) {
+        // Parse task module responses
+        while( cJSON_GetArraySize(task_module_response) && (error == OS_SUCCESS) ) {
+            cJSON *task_response = cJSON_DetachItemFromArray(task_module_response, 0);
```

Comment:
These responses should be stored in a temporal array so if there is an error we discard all of them.

Reply:
Ok